### PR TITLE
fix: add timeout to subproc

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adds a timeout around the sbt sub process being spawned. 

#### How should this be manually tested?
reduce timeout and run system-tests, or link with CLI and run snyk on scala project

### CODE REVIEW

File: lib/sub-process.js [hunk 1]
```diff
@@ -17,6 +20,10 @@ module.exports.execute = (command, args, options) => {
     let stderr = '';
 
     const proc = childProcess.spawn(command, args, spawnOptions);
+    if (PROC_TIMEOUT !== 0) {
+      setTimeout(kill(proc.pid, fullCommand), PROC_TIMEOUT);
```
Comment (lili2311) [inline lib/sub-process.js]: looks good! Could you add a test that shows the timeout message?

Comment (orsagie) [reply lib/sub-process.js]: Added test

File: lib/sub-process.js [hunk 2]
```diff
@@ -36,12 +45,19 @@ module.exports.execute = (command, args, options) => {
...
       }
-      resolve(stdout);
+      resolve(out.stdout);
     });
   });
 };
+
+function kill(id, out) {
+  return () => {
+    out.stderr = out.stderr + 'Process timed out. To set longer timeout run with `PROC_TIMEOUT=value_in_ms`';
```
Comment (kyegupov) [inline lib/sub-process.js]: maybe + '\n...' ?

Comment (orsagie) [reply lib/sub-process.js]: added

**Pull request merged**